File: leetcode/python/0225_implement_stack_using_queues.py
class MyStack(object):

  
    def __init__(self):
        self.queue = deque()
        # append to the back
        # pop to pop from the front
        # appendleft to append to the front
        # popleft to pop from the back
        # 1 2 3 4

    def push(self, x):
        """
        :type x: int
        :rtype: None
        """
        self.queue.append(x)
        
        for i in range(len(self.queue) - 1):
            self.queue.append(self.queue.pop())

    def pop(self):
        """
        :rtype: int
        """
        return self.queue.pop()

    def top(self):
        """
        :rtype: int
        """
        return self.queue[-1]

    def empty(self):
        """
        :rtype: bool
        """
        return len(self.queue) == 0


# Your MyStack object will be instantiated and called as such:
# obj = MyStack()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.top()
# param_4 = obj.empty()

File: leetcode/java/0225_implement_stack_using_queues.java
class MyStack {
    Queue<Integer> q = new LinkedList<>();

    public MyStack() {
    
    }
    
    public void push(int x) {
        if (q.size() == 0) {
            q.add(x);
            return;
        }
        q.add(x);
        for (int i = 0; i < q.size() - 1; i++) {
            int tmp = q.remove();
            q.add(tmp);
        }
    }
    
    public int pop() {
        return q.remove();
    }
    
    public int top() {
        return q.peek();
    }
    
    public boolean empty() {
        return q.isEmpty();
    }
}

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack obj = new MyStack();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.top();
 * boolean param_4 = obj.empty();
 */